base/bin/build.sh

```diff
@@ -1,27 +1,188 @@
+  echo ""
+  echo "                       If not specified or if the value \"${_DEFAULT}\" is specified, the image "
+  echo "                       will not be given the 'staging' tag."
+  echo "  ${_VERBOSE_OPTION}            Causes details to be printed as this command is run"
+  echo "  ${_HELP_OPTION}               Displays this help"
   exit 1
+}
+
+DOCKER_NAMESPACE="${_DEFAULT}"
+CANDIDATE_NAME="${_DEFAULT}"
+NODE_VERSION_TAG="${_DEFAULT}"
+UPLOAD_TO_STAGING="${_DEFAULT}"
+
+VERBOSE="false"
+RUNTIME_NAME="nodejs"
+
+# Parse the command line arguments
+while [ $# -gt 0 ]; do
+  case "$1" in
+    ${_NODE_VERSION_OPTION})
+      shift
+      if [ -z "${1}" ]; then
+        fail "${_NODE_VERSION_OPTION} must be followed by a version string"
+      fi
+      NODE_VERSION_TAG="${1}"
+      shift
+      ;;
+    ${_UPLOAD_TO_STAGING_OPTION})
+      shift
+      if [ "${1}" != "true" ] && [ "${1}" != "false" ]; then
+        fail "${_UPLOAD_TO_STAGING_OPTION} must be followed by either true or false"
+      fi
+      UPLOAD_TO_STAGING="${1}"
+      shift
+      ;;
+    ${_DOCKER_NAMESPACE_OPTION})
+      shift
+      if [ -z "${1}" ]; then
+        fail "${_DOCKER_NAMESPACE_OPTION} must be followind by a string"
+      fi
+      DOCKER_NAMESPACE="${1}"
+      shift
+      ;;
+    ${_CANDIDATE_NAME_OPTION})
+      shift
+      if [ -z "${1}" ]; then
+        fail "${_CANDIDATE_NAME_OPTION} must be followed by a string"
+      fi
+      CANDIDATE_NAME="${1}"
+      shift
+      ;;
+    ${_VERBOSE_OPTION})
+      shift
+      VERBOSE="true"
+      ;;
+    ${_HELP_OPTION})
+      shift
+      usage
+      ;;
+    *)
+      echo "Unknown option ${1}"
+      echo ""
+      shift
+      usage
+      ;;
+  esac
+done
+
+if [ "${DOCKER_NAMESPACE}" = "${_DEFAULT}" ]; then
+  DOCKER_NAMESPACE="gcr.io/$(gcloud config get-value project)"
+  print "A namespace was not specified with ${_DOCKER_NAMESPACE_OPTION}."
+  print "The value ${DOCKER_NAMESPACE} will be used."
+  print ""
+fi
+
+if [ "${CANDIDATE_NAME}" = "${_DEFAULT}" ]; then
+  CANDIDATE_NAME=$(date +%Y-%m-%d_%H_%M)
+  print "A candidate name was not specified with ${_CANDIDATE_NAME_OPTION}."
+  print "The value ${CANDIDATE_NAME} will be used."
+  print ""
+fi
+
+if [ "${NODE_VERSION_TAG}" = "${_DEFAULT}" ]; then
+  print "The Node.js version was not specified with the ${_NODE_VERSION_OPTION}."
+  print "The image will not be tagged with a Node.js version."
+  print ""
+fi
+
+if [ "${UPLOAD_TO_STAGING}" = "${_DEFAULT}" ]; then
+  UPLOAD_TO_STAGING="false"
+  print "Whether to specify the staging tag was not specified with ${_UPLOAD_TO_STAGING_OPTION}."
+  print "The image will not be tagged with the 'staging' tag."
+  print ""
 fi
 
 export IMAGE="${DOCKER_NAMESPACE}/${RUNTIME_NAME}:${CANDIDATE_NAME}"
 
+print "gcloud --version"
+if [ "${VERBOSE}" = "true" ]; then
+  gcloud --version
+fi
+print ""
+
+print "gcloud docker -- version"
+if [ "${VERBOSE}" = "true" ]; then
```

Comment:
Good catch.  I've updated it to use the newer syntax `print "$(gcloud docker --version)"`.